runtime/onert/core/src/compiler/Compiler.cc

```diff
@@ -671,8 +671,13 @@ std::vector<std::shared_ptr<CompilerArtifact>> Compiler::compile(const char *pac
     num_graphs = np.asUInt();
     for (uint32_t i = 0; i < (uint32_t)map.size(); ++i)
     {
-      options.partial_graph_options.index_to_graph[ir::OperationIndex{i}] =
-        ir::SubgraphIndex{map[i].asUInt()};
+      auto subg_idx = map[i].asUInt();
+      if (subg_idx > ir::SubgraphIndex::max())
+        throw std::runtime_error("The number of partition cannot exceed " +
+                                 std::to_string(ir::SubgraphIndex::max() + 1));
+      else
```

Comment:
Just for note

`else` is not necessary. But I would like to make sure to make static analysis tool happy, which complains about the possibility of overflow.